addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/discovery/SamsungTvDiscoveryParticipant.java

```diff
@@ -78,29 +91,26 @@ private String getLabel(RemoteDevice device) {
 
     @Override
     public ThingUID getThingUID(RemoteDevice device) {
-        if (device != null) {
+        String manufacturer = device.getDetails().getManufacturerDetails().getManufacturer();
+        String modelName = device.getDetails().getModelDetails().getModelName();
+        String friendlyName = device.getDetails().getFriendlyName();
 
-            String manufacturer = device.getDetails().getManufacturerDetails().getManufacturer();
-            String modelName = device.getDetails().getModelDetails().getModelName();
-            String friendlyName = device.getDetails().getFriendlyName();
+        if (manufacturer != null && modelName != null) {
 
-            if (manufacturer != null && modelName != null) {
+            if (manufacturer.toUpperCase().contains("SAMSUNG ELECTRONICS")) {
 
-                if (manufacturer.toUpperCase().contains("SAMSUNG ELECTRONICS")) {
+                // One Samsung TV contains several UPnP devices.
+                // Create unique Samsung TV thing for every MediaRenderer
+                // device and ignore rest of the UPnP devices.
 
+                if (device.getType().getType().equals("MediaRenderer")) {
```

Comment:
It's a best practice to always put all constants on the left hand site of equals comparisons to prevent NPEs.